app/models/drug_stock.rb

```diff
@@ -7,8 +7,15 @@ class DrugStock < ApplicationRecord
   validates :received, numericality: true, allow_nil: true
   validates :for_end_of_month, presence: true
 
+  def self.latest_for_facilities_grouped_by_protocol_drug(facilities, end_of_month)
+    drug_stock_list = latest_for_facilities(facilities, end_of_month) || []
+    drug_stock_list.each_with_object({}) { |drug_stock, acc|
+      acc[drug_stock.protocol_drug.id] = drug_stock
+    }
+  end
```

Comment:
Would be cool to find a way to reuse a function for this and `DrugStocksQuery#drug_stocks_by_drug_id`. It's not too important though.